Examples/Install-Win32OpenSSH/Install-Win32OpenSSH.ps1

```diff
@@ -0,0 +1,178 @@
+    ### Create registry configuration for SSH default shell
+    New-Item -Path HKLM:\Software\OpenSSH -ErrorAction Ignore
+    Set-ItemProperty -Path HKLM:\SOFTWARE\OpenSSH -Name DefaultShell -Value c:\pwsh\pwsh.exe
+    #Set-ItemProperty -Path HKLM:\SOFTWARE\OpenSSH -Name DefaultShellCommandOption -Value '-Command'
+}
+ 
+function Complete-SSHDConfig {
+  Write-Log -Message 'Configuring SSH daemon'
+ 
+  $ConfigPath = '{0}\ssh\sshd_config' -f $env:ProgramData
+    
+  ### Generate a Certificate Authority key to sign user public keys with
+  ./ssh-keygen -t rsa -f ca_userkeys -q -N '""'    
+  Add-Content -Path $ConfigPath -Value "`r`n",'TrustedUserCAKeys ca_userkeys.pub', 'PubkeyAuthentication yes'
+  Write-Log -Message 'Finished generating Certificate Authority signing key'
+  
+  Add-Content -Path $ConfigPath -Value '','Subsystem powershell c:/pwsh/pwsh.exe -sshs -NoLogo -NoProfile'
+  $NewConfig = Get-Content -Path $ConfigPath | ForEach-Object -Process {
+    $PSItem -replace '^.*PasswordAuthentication.*$', 'PasswordAuthentication No'
+  }
+  Set-Content -Path $ConfigPath -Value $NewConfig
+  Write-Log -Message 'Finished configuring SSH daemon'
+}
+ 
+function Build-SSHDAuthorizedKeys {
+  <#
+  .Parameter ParameterName
+  The name of the AWS Systems Manager parameter that contains your SSH public key.
+  #>
+  [CmdletBinding()]
+  param (
+  )
+ 
+  Write-Log -Message 'Configuring authorized_keys file'
+  $ProfilePath = '{0}\Users\Administrator\.ssh' -f $env:SystemDrive
+  if (!(Test-Path -Path $ProfilePath)) {
+    mkdir -Path $ProfilePath
+  }
+ 
+  # Write the user's public key into the Administrator account's profile 
+  $PublicKey = Invoke-RestMethod -Uri http://169.254.169.254/latest/meta-data/public-keys/0/openssh-key
+  Set-Content -Path $ProfilePath\authorized_keys -Value $PublicKey
+  Write-Log -Message 'Finished configuring authorized_keys file'
+}
+ 
+function StartServices {
+  [CmdletBinding()]
+  param ( )
+ 
+  Write-Log -Message 'Configuring services for automatic startup'
+  Set-Service sshd -StartupType Automatic -ErrorAction Ignore
+  Set-Service ssh-agent -StartupType Automatic -ErrorAction Ignore
+ 
+  Write-Log -Message 'Starting services'
+  Start-Service -Name sshd
+  Start-Service -Name ssh-agent
+  Write-Log -Message 'Finished starting services'
+}
+ 
+function Set-OpenSSHFirewall {
+  [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Low')]
+  param (
+    [switch] $Force
+  )
+  if ($PSCmdlet.ShouldProcess('Windows Firewall', 'Open SSH connectivity')) {
+    if ($Force -or $PSCmdlet.ShouldContinue()) {
+      Write-Log -Message 'Starting configuration of Windows Firewall rule for SSH daemon'
+      New-NetFirewallRule -LocalPort 22 -Direction Inbound -Profile Any -DisplayName sshd22 -Name 'Secure Shell (SSH) Daemon (sshd)' -Enabled true -Action Allow -Protocol TCP -ErrorAction Ignore
+      Write-Log -Message 'Configured the Windows Firewall for ssh daemon'
+    }
+  }
+}
+ 
+Install-PowerShellCore
```

Comment:
Why install PowerShell Core and Putty, when the script is intended to install only SSH? Are these required?

Reply:
It's not intended to only install OpenSSH. It installs PowerShell Core, Win32-OpenSSH, and configures the SSH daemon and Windows Firewall. PuTTY is a troubleshooting tool in case any issues occur.